keras_nlp/samplers/beam_sampler.py

```diff
@@ -101,6 +102,8 @@ def __init__(
         return_all_beams=False,
         **kwargs,
     ):
+        assert_tf_backend(self.__class__.__name__)
```

Comment:
maybe add a comment "Beam search only works in Tensorflow backend + short explanation"?